src/openassetio-python/cmodule/src/ContextBinding.cpp

```diff
@@ -14,15 +14,21 @@ void registerContext(const py::module& mod) {
   using openassetio::Context;
   using openassetio::ContextPtr;
   using openassetio::managerApi::ManagerStateBasePtr;
-  using openassetio::trait::TraitsDataPtr;
+  using openassetio::trait::TraitsData;
+  using PyRetainingTraitsDataPtr =
+      openassetio::PyRetainingSharedPtr<openassetio::trait::TraitsData>;
   using PyRetainingManagerStateBasePtr =
       openassetio::PyRetainingSharedPtr<openassetio::managerApi::ManagerStateBase>;
 
   py::class_<Context, ContextPtr> context{mod, "Context", py::is_final()};
 
   context
-      .def(py::init(RetainCommonPyArgs::forFn<&Context::make>()),
-           py::arg_v("locale", TraitsDataPtr{}), py::arg_v("managerState", ManagerStateBasePtr{}))
+      .def(py::init(
+               [](PyRetainingTraitsDataPtr locale, PyRetainingManagerStateBasePtr managerState) {
```

Comment:
[fix] Can we note why were doing this (its in the commit, but would be good here too)

Reply:
Added a comment in 5eaa8493